src/main/logseq/api/block.cljs

```diff
@@ -35,19 +35,39 @@
     (keyword :logseq.property property-name)
     property-name))
 
+(defn sanitize-user-property-name
+  [k]
+  (if (string? k)
+    (-> k (string/trim)
+      (string/replace #"^[:_]+" "")
+      (string/lower-case))
+    k))
+
+(defn resolve-property-prefix-for-db
+  [^js plugin]
+  (when (some-> js/window.LSPlugin (.-PluginLocal))
+    (or (some->> plugin (.-id) (sanitize-user-property-name) (str "."))
+      "._api")))
+
 ;; FIXME: This ns should not be creating idents. This allows for ident conflicts
 ;; and assumes that names directly map to idents which is incorrect and breaks for multiple
 ;; cases e.g. a property that has been renamed or sanitized. Instead it should
 ;; find a property's ident by looking up the property in the db by its title
 (defn get-db-ident-for-user-property-name
   "Finds a property :db/ident for a given property name"
-  [property-name]
-  (let [property-name' (if (string? property-name)
-                         (keyword property-name) property-name)
-        property-name' (convert?to-built-in-property-name property-name')]
-    (if (qualified-keyword? property-name')
-      property-name'
-      (keyword "plugin.property" (encode-user-property-name property-name)))))
+  ([property-name] (get-db-ident-for-user-property-name property-name nil))
+  ([property-name prefix]
+   (let [property-name' (if (string? property-name)
+                          (keyword property-name) property-name)
+         property-name' (convert?to-built-in-property-name property-name')]
+     (if (qualified-keyword? property-name')
+       property-name'
+       (keyword (str "plugin.property" prefix) (encode-user-property-name property-name))))))
```

Comment:
```suggestion
       (db-ident/create-db-ident-from-name (str "plugin.property" prefix) (encode-user-property-name property-name)))))
```
If this fn is being used to create idents, we should be using an existing fn that creates valid EDN keywords and is well tested. The current sanitization allows for a lot of buggy keywords as described in https://clojure.org/reference/reader#_symbols